### DISCUSSION TITLE
Limitation in auto completions list

### DISCUSSION THREAD

**Discussion by Bouh** [4ian/GDevelop#2634] [Archive] [answered]
A limitation was introduced in the auto-completion list when typing an expression.

The list is limited to 25 expressions, for which reason?

The expression builder exist, but it shouldn't be a reason to prevented people to search in this list.
[Source](https://forum.gdevelop-app.com/t/how-do-i-access-these-additional-expressions/32534)

![image](https://user-images.githubusercontent.com/1670670/117802943-58b32880-b256-11eb-8f68-8d9b64b288bc.png)
![image](https://user-images.githubusercontent.com/1670670/117802990-65378100-b256-11eb-845f-cfd1af9c6b02.png)

**Comment by 4ian** [comment, marked as answer]
This list is not virtualised, so without a maximum number, typing a single character like "a" could return a lot of results (200, more?) that would end up slowing down the rendering (basically, you would have a lag, especially visible on low end devices, because the browser has to handle these 200+ DOM nodes).
This means that while you type, you would get visible slow downs => not a great experience. In the future, things like "concurrent rendering" in React could solve this (allowing you to continue typing, and React will stop rendering if it's too long).
We could also virtualise the results, like in the object list or in the search results of the asset store/extensions.

EDIT: we could also add more items when we detect we reach the bottom of the list :)

**Reply by tristanbob** [reply]
Would the quickest/easiest fix be what 4ian described in his edit? 
 
Add more expressions once the user has scrolled to the bottom.
